### PR TITLE
Considering attribute projection when choosing index for AthenaDynamoDBConnector.

### PR DESCRIPTION
**Pull request by mchlsun**
*Issue #, if available: #279

*Description of changes: adding validation to check if projected attributes of an index contails all required attributes of a query when choosing index in AthenaDynamoDBConnector. Fix to issue#279

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Review by avirtuos — APPROVED** [review]
This looks good to me. Thanks for the contribution. Before merging, I'm going to ask @atennak1 to take a look since he is more familiar with this connector than I am. Once he takes a look I'm hopeful we can merge it.

**Comment by avirtuos** [comment]
@atennak1 This looks good to me but I'm somewhat unclear on why this didn't work already since much of the code already seemed to exist. Is the main delta that your existing implementation didn't account for 'covered' indexes? Anyway, can you take a look before we merge this? Thanks

**Comment by avirtuos** [comment]
@mchlsun Thanks for the contribution. Can you share some details of how this got on your radar?

File: athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTableUtils.java [hunk 1]
```diff
@@ -88,17 +91,19 @@ public static DynamoDBTable getTable(String tableName, ThrottlingInvoker invoker
...
-        });
-        globalSecondaryIndexes.forEach(i -> {
-            KeyNames indexKeys = getKeys(i.getKeySchema());
-            indices.add(new DynamoDBTable(i.getIndexName(), indexKeys.getHashKey(), indexKeys.getRangeKey(), table.getAttributeDefinitions(), ImmutableList.of(), i.getIndexSizeBytes(), i.getItemCount(),
-                    i.getProvisionedThroughput() != null ? i.getProvisionedThroughput().getReadCapacityUnits() : PSUEDO_CAPACITY_FOR_ON_DEMAND));
+            // DynamoDB automatically fetches all attributes from the table for local secondary index, so ignore projected attributes
+            indices.add(new DynamoDBIndex(i.getIndexName(), indexKeys.getHashKey(), indexKeys.getRangeKey(), ProjectionType.ALL, ImmutableList.of()));
         });
+        globalSecondaryIndexes.stream()
+              .filter(i -> IndexStatus.fromValue(i.getIndexStatus()).equals(IndexStatus.ACTIVE))
```
Comment (atennak1) [inline athena-dynamodb/src/main/java/com/amazonaws/athena/connectors/dynamodb/util/DDBTableUtils.java:101]: nice add

**Review by atennak1 — APPROVED** [review]
Looks good and thanks for the additional tests

**Comment by atennak1** [comment]
> @atennak1 This looks good to me but I'm somewhat unclear on why this didn't work already since much of the code already seemed to exist. Is the main delta that your existing implementation didn't account for 'covered' indexes? Anyway, can you take a look before we merge this? Thanks

Yep I knew about this usecase but I guess I forgot to account for it

**Pull request merged**